src/runtime/app.ts

```diff
@@ -95,33 +104,39 @@ export function create(): App {
   const appState = createAppState()
   const serverComponent = Server.create(appState)
   const schemaComponent = Schema.create(appState)
-  const settingsComponent = Settings.create({
+  const settingsComponent = Settings.create(appState, {
     serverSettings: serverComponent.private.settings,
     schemaSettings: schemaComponent.private.settings,
     log,
   })
-  const api: PrivateApp = {
+  const api: App = {
     log: log,
     settings: settingsComponent.public,
     schema: schemaComponent.public,
     server: serverComponent.public,
-    // todo call this in the start module
     assemble() {
       if (appState.assembled) return
 
-      appState.assembled = true
+      appState.assembled = {} as any
```

Comment:
Couldn't you use a `Partial` here?

Reply:
Leaving a todo for now 

![image](https://user-images.githubusercontent.com/284476/81171791-05cfd280-8f6b-11ea-8bba-6a947a144032.png)


Think the settings readonly is causing issue. Tried Mutable modifier from type fest (also related: https://github.com/microsoft/TypeScript/issues/24509) but didn't seem to help.